workerapi/server/server.go

```diff
@@ -31,63 +32,83 @@ func MakeServer(
 		protocolFactory)
 }
 
+// defining StatsCollectInterval type so it can be injected via ICE
+// (avoid conflict with other injected integers)
+type StatsCollectInterval int
+
 type handler struct {
-	stat         stats.StatsReceiver
-	run          runner.Service
-	timeLastRpc  time.Time
-	mu           sync.Mutex
-	currentCmd   *runner.Command
-	currentRunID runner.RunID
+	stat                 stats.StatsReceiver
+	run                  runner.Service
+	timeLastRpc          time.Time
+	mu                   sync.RWMutex
+	currentCmd           *runner.Command
+	currentRunID         runner.RunID
+	statsCollectInterval StatsCollectInterval
 }
 
 // Creates a new Handler which combines a runner.Service to do work and a StatsReceiver
-func NewHandler(stat stats.StatsReceiver, run runner.Service) worker.Worker {
+func NewHandler(stat stats.StatsReceiver, run runner.Service, idtCh snapshot.InitDoneTimeCh, statsCollectInterval StatsCollectInterval) worker.Worker {
 	scopedStat := stat.Scope("handler")
-	h := &handler{stat: scopedStat, run: run}
-	go h.stats()
+	h := &handler{stat: scopedStat, run: run, timeLastRpc: time.Now(), statsCollectInterval: statsCollectInterval}
+	go h.stats(idtCh)
 	return h
 }
 
 // Periodically output stats
 //TODO: runner should eventually be extended to support stats, multiple runs, etc. (replacing loop here).
-func (h *handler) stats() {
-	var startTime time.Time
+func (h *handler) stats(idtCh snapshot.InitDoneTimeCh) {
+	var startTime time.Time = time.Now()
+	var initTime time.Duration
 	nilTime := time.Time{}
-	ticker := time.NewTicker(time.Millisecond * time.Duration(500))
+	initDoneTime := nilTime
+	ticker := time.NewTicker(time.Millisecond * time.Duration(h.statsCollectInterval))
```

Comment:
You should define statsCollectInterval as a time.Duration and convert from int->Duration when we define the value main.go.